server/routerlicious/packages/services/src/tenant.ts

```diff
@@ -97,10 +97,10 @@ export class TenantManager implements core.ITenantManager {
     }
 
     public async getKey(tenantId: string, includeDisabledTenant = false): Promise<string> {
-        const result = await Axios.get(
-            `${this.endpoint}/api/tenants/${encodeURIComponent(tenantId)}/key`,
+        const result = await Axios.get<core.ITenantKeys>(
```

Comment:
This getKey function is to get a valid tenant key to generate token, and put token in request header to send requests to other internal microservices. So in theory we can return either key1 or key2 since later in Riddler we will validate the token with either key1 or key2. All new created tenants will have 2 keys, but current existing tenants will only have key1 unless they refresh key2. So we just use key1 here since all tenants have key1 all the time. Same logic applies to BYOS, we use key1 to generate token and send requests to historian.

Reply:
I think we can think of it this way. Right now at this moment all tenants have 2 keys, key1 is a valid key, key2 is an empty string. So we are using key1 right now. Later on some of the tenants will have a valid key2, but some of the tenants won't. So we will still use key1.

Reply:
Makes sense. Is there any scenario where `key1` is not valid, so instead we could check if `key2` is there and use it in that case to avoid a possible error down the line?

Reply:
Yeah I also thought about it. There won't be a scenario key1 is not valid. It would be the same logic happening right now with the only key. We read from tenant db to get key1, generate jwt, send request, Riddler reads key1 from tenant db, validates token with key1. It's a close loop so there and all tenants have valid key1 so there won't be any invalid scenario.

Reply:
Awesome - thanks for clarifyin! 😃